src/openms/include/OpenMS/ANALYSIS/MAPMATCHING/QTClusterFinder.h

```diff
@@ -146,13 +151,83 @@ namespace OpenMS
     /// Sets algorithm parameters
     void setParameters_(double max_intensity, double max_mz);
 
-    /// Generates a consensus feature from the best cluster and updates the clustering
-    void makeConsensusFeature_(std::list<QTCluster>& clustering,
+    /**
+     * @brief Extract the best cluster from cluster_heads and turn it into a consensus feature
+     * 
+     * @param cluster_heads the heap where the clusters are stored, must not be empty
+     * @param feature the resulting consensus feature is constructed in this variable
+     * @param element_mapping the element mapping is used to update clusters when features are removed
+     * @param grid the grid is used to find new features for clusters that have to be updated
+     * @param handles used to access clusters if we know their id from the element mapping
+     * 
+     * @return bool whether a consensus feature was made or not
+     */
+    bool makeConsensusFeature_(Heap& cluster_heads,
                                ConsensusFeature& feature,
-                               ElementMapping& element_mapping, Grid&);
+                               ElementMapping& element_mapping,
+                               Grid& grid,
+                               const std::vector<Heap::handle_type>& handles);
 
-    /// Computes an initial QT clustering of the points in the hash grid
-    void computeClustering_(Grid& grid, std::list<QTCluster>& clustering);
+    /**
+     * @brief Computes an initial QT clustering of the points in the hash grid
+     * 
+     * @param grid the grid is used to find new features for clusters that have to be updated
+     * @param cluster_heads the heap where the QTClusters are inserted
+     * @param cluster_data vector where the BulkData objects of the clusters are stored
+     * @param handles vector where handles of the inserted clusters are stored
+     * @param element_mapping the element mapping where all the clusters get registered for their features
+     */
+    void computeClustering_(Grid& grid,
+                            Heap& cluster_heads,
+                            std::vector<QTCluster::BulkData>& cluster_data,
+                            std::vector<Heap::handle_type>& handles,
+                            ElementMapping& element_mapping);
+
+    /** 
+     * @brief Removes id of current top cluster in the heap from element mapping
+     * 
+     * @param cluster the current top cluster in the heap which id is removed
+     * @param element_mapping the element mapping from which the ids are removed
+     */
+    void removeFromElementMapping_(const QTCluster& cluster,
+                                   ElementMapping& element_mapping);
+    
+    /** 
+     * @brief creates a consensus feature from the given elements
+     * 
+     * @param feature the resulting consensus feature is constructed in this variable
```

Comment:
`@param[out] feature The resulting consensus feature is constructed here

Reply:
done